satpy/writers/__init__.py

```diff
@@ -865,14 +865,17 @@ def separate_init_kwargs(cls, kwargs):
         return init_kwargs, kwargs
 
     def save_dataset(self, dataset, filename=None, fill_value=None,
-                     overlay=None, decorate=None, compute=True, **kwargs):
+                     overlay=None, decorate=None, compute=True, units=None, **kwargs):
         """Save the ``dataset`` to a given ``filename``.
 
         This method creates an enhanced image using :func:`get_enhanced_image`.
         The image is then passed to :meth:`save_image`. See both of these
         functions for more details on the arguments passed to this method.
 
         """
+        if units is not None:
+            import pint_xarray  # noqa
+            dataset = dataset.pint.quantify().pint.to(units)
```

Comment:
Not sure if you wanted this PR reviewed yet, but do you plan on adding error handling? Both for units being specified but pint not being available and for the case where the units can't be converted.

Oh and maybe `units` could be a dictionary? Mapping input units to output units? I'm thinking of the `save_datasets` case where you have visible and IR bands or any number of level 2 products.

Reply:
> I haven't planned for the multiple saving case. This PR is just for save_dataset

...but...but...it would benefit my projects :cry:

Reply:
I understand, but this will require modification of the other function (`save_datasets`) and other tests. I suggest we do this in another PR.

Reply:
Yeah understood. Just complaining for the sake of complaining.